src/util/sandbox.cpp

```diff
@@ -369,80 +369,119 @@ QString Sandbox::migrateOldSettings() {
     QString title = QObject::tr("Upgrading old Mixxx settings");
-    QString oldPath = homePath + QLatin1String("/Library/Application Support/Mixxx");
     QMessageBox::information(nullptr,
             title,
             QObject::tr(
-                    "Due to Mac sandboxing, Mixxx needs your permission to "
-                    "access your music library "
-                    "and settings from Mixxx versions before 2.3.0. After "
-                    "clicking OK, you will see a file picker. "
-                    "To give Mixxx permission, press the Ok button in the file "
-                    "picker."
+                    "Due to macOS sandboxing, Mixxx needs your permission "
+                    "to access your music library and settings from Mixxx "
+                    "versions before 2.3.0. After clicking OK, you will see a "
+                    "file selection dialog. "
                     "\n\n"
-                    "If you do not want to grant Mixxx access click Cancel on "
-                    "the file picker."));
+                    "To allow Mixxx to use your old library and settings, "
+                    "click the Open button in the file selection dialog. "
+                    "Mixxx will then move your old settings into the sandbox. "
+                    "This only needs to be done once."
+                    "\n\n"
+                    "If you do not want to grant Mixxx access, click Cancel "
+                    "on the file picker."));
+
     QString result = QFileDialog::getExistingDirectory(
             nullptr,
             title,
-            oldPath);
-    if (result != oldPath) {
-        qInfo() << "Sandbox::migrateOldSettings: User declined to migrate "
-                   "old settings from"
-                << oldPath << "User selected" << result;
+            legacySettingsPath);
+    if (result != legacySettingsPath) {
+        qInfo() << "Sandbox::migrateOldSettings: User declined to migrate old settings from"
+                << legacySettingsPath << "User selected" << result;
         return sandboxedPath;
     }
 
-    // Sandbox::askForAccess cannot be used here because it depends on settings being
-    // initialized. There is no need to store the bookmark anyway because this is a
-    // one time process.
 #ifdef __APPLE__
     CFURLRef url = CFURLCreateWithFileSystemPath(
-            kCFAllocatorDefault, QStringToCFString(oldPath), kCFURLPOSIXPathStyle, true);
+            kCFAllocatorDefault, QStringToCFString(legacySettingsPath), kCFURLPOSIXPathStyle, true);
     if (url) {
         CFErrorRef error = NULL;
-        CFDataRef bookmark = CFURLCreateBookmarkData(
-                kCFAllocatorDefault,
-                url,
-                kCFURLBookmarkCreationWithSecurityScope,
-                nil,
-                nil,
-                &error);
-        CFRelease(url);
-        if (bookmark) {
-            QFile oldSettings(oldPath);
+        if (s_bInSandbox) {
```

Comment:
This requires `Sandbox::checkSandboxed` to be called, which is not really apparent. Idea: Make `s_bIsSandboxed` a `std::optional`. If it doesn't hold a value, run `s_bIsStandboxed = Sandbox::isSandboxed()` (change `void Sandbox::checkSandbox()` to `bool Sandbox::isSandboxed` and let it return the value instead of assigning the singleton).

Reply:
Sandbox::checkSandbox is called before this.

Reply:
Yes, but it's not obvious if someone touches the code later. I think `std::optional` would be helpful, or at least add a comment.

Reply:
I could just call Sandbox::checkSandbox at the beginning of this function, but IMO that would make the startup code a bit harder to follow.

Reply:
On further thought, the call to `Sandbox::checkSandboxed()` needs to stay where it is because `Sandbox::migrateOldSettings` is only invoked if the `--settingsPath` command line argument is not used.